DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs

```diff
@@ -0,0 +1,177 @@
+            //Then
+            _courseRepositoryMock.Verify(x => x.DeleteAllTopicsByCourseId(givenCourseId), Times.Once);
+            _topicRepositoryMock.Verify(x => x.AddTopicsToCourse(givenTopicsToUpdate), Times.Once);
+        }
+        [Test]
+        public void UpdateCourseTopicsByCourseId_TopicsInDatabaseAreAbsentForCourse_AddedTopicsForCourse()
+        {
+            //Given
+            var givenCourseId = 3;
+            var givenTopicsToUpdate = CourseData.GetListCourseTopicDto(3);
+            var toicsFromDB = CourseData.GetListCourseTopicDto(5);
+            _courseRepositoryMock.Setup(x => x.SelectAllTopicsByCourseId(givenCourseId)).Returns(toicsFromDB);
+            _courseRepositoryMock.Setup(x => x.UpdateCourseTopicsByCourseId(givenTopicsToUpdate));
+            var sut = new CourseService(_topicRepositoryMock.Object, _courseRepositoryMock.Object);
+            //When
+            sut.UpdateCourseTopicsByCourseId(givenCourseId, givenTopicsToUpdate);
+            //Then
+            _topicRepositoryMock.Verify(x => x.AddTopicsToCourse(givenTopicsToUpdate), Times.Once);
+        }
+        [Test]
+        public void UpdateCourseTopicsByCourseId_TopicsCountIsZero_Returnerd()
+        {
+            //Given
+            var givenCourseId = 3;
+            var givenTopicsToUpdate = CourseData.GetListCourseTopicDto(5);
+            var toicsFromDB = CourseData.GetListCourseTopicDto(3);
+            _courseRepositoryMock.Setup(x => x.SelectAllTopicsByCourseId(givenCourseId)).Returns(toicsFromDB);
+            _courseRepositoryMock.Setup(x => x.UpdateCourseTopicsByCourseId(givenTopicsToUpdate));
+            var sut = new CourseService(_topicRepositoryMock.Object, _courseRepositoryMock.Object);
+            //When
+            sut.UpdateCourseTopicsByCourseId(givenCourseId, givenTopicsToUpdate);
+            //Then
+            _courseRepositoryMock.Verify(x => x.UpdateCourseTopicsByCourseId(givenTopicsToUpdate), Times.Never);
+        }
+        [Test]
+        public void UpdateCourseTopicsByCourseId_PositionsNotUniqueness_ThrownException()
+        {
+            //Given
+            var givenCourseId = 3;
+            var givenTopicsToUpdate = CourseData.GetListCourseTopicDto(6);
+            var toicsFromDB = CourseData.GetListCourseTopicDto(3);
+            _courseRepositoryMock.Setup(x => x.SelectAllTopicsByCourseId(givenCourseId)).Returns(toicsFromDB);
+            _courseRepositoryMock.Setup(x => x.UpdateCourseTopicsByCourseId(givenTopicsToUpdate));
+            var sut = new CourseService(_topicRepositoryMock.Object, _courseRepositoryMock.Object);
+            //When
+            var exception = Assert.Throws<ValidationException>(() => 
+            sut.UpdateCourseTopicsByCourseId(givenCourseId, givenTopicsToUpdate));
+            //Then
+            Assert.That(exception.Message, Is.EqualTo("the same positions of topics in the course"));
```

Comment:
please move this message to a const
(and use this constant in the service method as well)